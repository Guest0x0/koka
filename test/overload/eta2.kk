noinline fun plus(x:int,y:int,z:int=1)
  x+y+z

noinline fun foo( xs : list<a>, zero : a, ?plus : (a,a) -> a) : a
  xs.foldl(zero,?plus)

pub fun main()
  [1,2,3].foo(0)