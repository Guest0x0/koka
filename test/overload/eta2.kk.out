9
 
overload/eta2/foo: forall<a> (xs : list<a>, zero : a, ?plus : (a, a) -> a) -> a
overload/eta2/main: () -> int
overload/eta2/plus: (x : int, y : int, z : ? int) -> int